test/unit/domain_test.rb

```diff
@@ -20,6 +20,14 @@ def setup
     assert !other_domain.save
   end
 
+  test "should have a unique case-insensitive name" do
+    other_domain = Domain.new(:name => @domain.name.upcase)
+    refute other_domain.save
+    assert_raises ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid do
```

Comment:
Removing the case-insensitive enforcement from the database isn't the way to go as you can encounter this situation when 2 people submit a domain to be created or updated.
```
1. USER1 - Query the database to see if this domain already exists.
2. USER2 - Query the database to see if this domain already exists.
3. USER1 - OK, doesn't exist, let's insert the record.
4. USER2 - OK, doesn't exist, let's insert the record.
```
Removing the test isn't good either IMO. We want to verify this behavior. I'm not sure right now how to move forward, I'm checking why exactly this happens, even when not loading the test via rake (`ruby -I'test' test/unit/domain_test.rb`)